Sources/Bot/Services/DispatchService.swift

```diff
@@ -80,29 +77,25 @@ public final class DispatchService {
 
     private func pullRequestDidChange(event: PullRequestEvent) {
         logger.log("📣 Pull Request did change \(event.pullRequestMetadata) with action `\(event.action)`")
+
         let targetBranch = event.pullRequestMetadata.reference.target.ref
-        let existingService = mergeServices.modify { (dict: inout [String: MergeService]) -> MergeService? in
+        let existingService = mergeServices.modify { (dict: inout [String: MergeService]) -> MergeService in
             if let service = dict[targetBranch] {
-                // If service was already existing, return it so we'll send the pullRequestChangesObserver event outside this `modify` below
                 return service
             } else {
-                dict[targetBranch] = makeMergeService(
+                let newService = makeMergeService(
                     targetBranch: targetBranch,
                     scheduler: self.scheduler,
                     initialPullRequests: [event.pullRequestMetadata.reference]
                 )
-                // If MergeService didn't exist yet and we just created it, return nil so that we DON'T send the event on pullRequestChangesObserver
-                // outside this `modify` below (because we already passed the PR to initialPullRequests parameters when creating the service – and
-                // the service would still be `.starting` and it would not be ready to receive those events anyway)
-                return nil
+                dict[targetBranch] = newService
+                return newService
             }
         }
-        existingService?.pullRequestChangesObserver.send(value: (event.pullRequestMetadata, event.action))
+        existingService.pullRequestChangesObserver.send(value: (event.pullRequestMetadata, event.action))
```

Comment:
Notice that we now send the event in all cases, and let the `MergeService` filter that event out (the `Feedback.pullRequestChanges` of `MergeService` already filters out those events if the service was just started, using `predicate: { $0.status != .starting }`).

Which means we don't leak logic and implementation details about the internals of `MergeService` to `DispatchService` anymore, it doesn't have to know that logic about "don't send event if service was just created", instead we dispatch in all cases and let the `MergeService` do the filtering